### PR TITLE
Configure gripper speed and effort with hardware interface

### PR DESCRIPTION
**Pull request by pac48**
Many robot grippers support effort, position, and velocity controls. The current technique to set the velocity involves reading a hardcoded value from the URDF. This is problematic when the user wants to change the gripper velocity at runtime. After discussions in the ros2_control WG meeting, the best strategy to achieve this goal while allowing backporting to Humble is to add a new controller that specifically supports this and deprecate the old controller. 

This PR adds a controller called `antipodal_gripper_controllers` which has parameters to optionally claim the hardware interfaces `JOINT_NAME/gripper_effort` and `JOINT_NAME/gripper_speed`, which will be used to set the gripper speed and effort if enabled.  

This PR requires:
- [x] depends on https://github.com/ros-controls/control_msgs/pull/99

### CODE REVIEW

**Comment by moriarty** [comment]
```
  --- stderr: antipodal_gripper_controller
  In file included from /home/runner/work/ros2_controllers/ros2_controllers/.work/target_ws/src/ros2_controllers/antipodal_gripper_controller/src/antipodal_gripper_action_controller.cpp:18:
  /home/runner/work/ros2_controllers/ros2_controllers/.work/target_ws/src/ros2_controllers/antipodal_gripper_controller/include/antipodal_gripper_controller/antipodal_gripper_action_controller.hpp:30:10: fatal error: control_msgs/action/antipodal_gripper_command.hpp: No such file or directory
     30 | #include "control_msgs/action/antipodal_gripper_command.hpp"
        |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

@pac48 can you include a temporary change on this pr to: 

https://github.com/ros-controls/ros2_controllers/blob/master/ros2_controllers-not-released.rolling.repos

I believe that should turn *Rolling Semi-Binary Build - main / semi_binary / rolling main (pull_request)* green and it will be require between ros syncs anyway

File: antipodal_gripper_controller/doc/userdoc.rst
Comment (MarqRazz) [inline antipodal_gripper_controller/doc/userdoc.rst:5]:
```suggestion
Antipodal Gripper Action Controller
```

File: antipodal_gripper_controller/doc/userdoc.rst
Comment (MarqRazz) [inline antipodal_gripper_controller/doc/userdoc.rst:8]:
```suggestion
Controller for executing a `AntipodalGripperCommand` action for simple antipodal grippers.
```

Robotiq 2F grippers are Antipodal and not `single-dof` even though you can command them with a single value.

**Review by MarqRazz — COMMENTED** [review]
I still need to pull this down and test but here are some high level suggestions.

File: antipodal_gripper_controller/test/test_antipodal_gripper_controller.cpp
Comment (pac48) [reply antipodal_gripper_controller/test/test_antipodal_gripper_controller.cpp:33]: I think it is needed here because it is defined inside a class in `antipodal_gripper_action_controller.hpp`

File: antipodal_gripper_controller/src/antipodal_gripper_action_controller_parameters.yaml
Comment (MarqRazz) [reply antipodal_gripper_controller/src/antipodal_gripper_action_controller_parameters.yaml:53]: We could also consider something like `enable_max_effort_interface`

File: antipodal_gripper_controller/src/antipodal_gripper_action_controller_parameters.yaml
Comment (MarqRazz) [reply antipodal_gripper_controller/src/antipodal_gripper_action_controller_parameters.yaml:28]: My main goal by renaming these is to not confuse users between these and the actual `velocity`/`effort` state/command interfaces

Comment (pac48) [reply antipodal_gripper_controller/src/antipodal_gripper_action_controller_parameters.yaml]: I added the target max_velocity` parameter, good catch!

Regarding your second comment, that is why I added the gripper prefix. The interfaces would show up with ros2_control add
```
command_interfaces:
  robotiq_85_left_knuckle_joint/gripper_speed
  robotiq_85_left_knuckle_joint/gripper_effort
```
As opposed to standard interfaces:  
```
command_interfaces:
  robotiq_85_left_knuckle_joint/velocity
  robotiq_85_left_knuckle_joint/effort
```
Because they are command interfaces, it seems prefixing with set_gripper could be redundant.

Comment (pac48) [reply antipodal_gripper_controller/src/antipodal_gripper_action_controller_parameters.yaml]: Okay, I change the interface names to `set_gripper_max_velocity` and `set_gripper_max_effort`

File: antipodal_gripper_controller/src/antipodal_gripper_action_controller_parameters.yaml
Comment (pac48) [reply antipodal_gripper_controller/src/antipodal_gripper_action_controller_parameters.yaml]: Okay, I changed it to `configure_max_` ...

File: gripper_controllers/include/gripper_controllers/gripper_action_controller_impl.hpp
Comment (MarqRazz) [inline gripper_controllers/include/gripper_controllers/gripper_action_controller_impl.hpp:43]: :1st_place_medal:

File: ros2_controllers-not-released.rolling.repos
Comment (MarqRazz) [inline ros2_controllers-not-released.rolling.repos:9]: Reminder: needs to be updated after the msgs are updated and this is merged.

**Review by MarqRazz — APPROVED** [review]
I think this is good to go after we add a little more documentation. @bmagyar have you had a chance to check this out?

File: antipodal_gripper_controller/src/antipodal_gripper_action_controller_parameters.yaml
Comment (saikishor) [inline antipodal_gripper_controller/src/antipodal_gripper_action_controller_parameters.yaml:29]: Just so I understand, this max velocity in m/sec should be an API available in the hardware driver. We don't compute and limit it right? Or did I understand wrong?

File: antipodal_gripper_controller/doc/userdoc.rst
Comment (saikishor) [reply antipodal_gripper_controller/doc/userdoc.rst:8]: Yes some more documentation would be helpful. It would also be nice to add some gripper manufacturers that are of antipodal type. It would be easier for the people to connect

File: antipodal_gripper_controller/src/antipodal_gripper_action_controller_parameters.yaml
Comment (saikishor) [inline antipodal_gripper_controller/src/antipodal_gripper_action_controller_parameters.yaml:5]:
```suggestion
    description: "Hz",
    read_only: true,
```

File: antipodal_gripper_controller/src/antipodal_gripper_action_controller_parameters.yaml
Comment (saikishor) [inline antipodal_gripper_controller/src/antipodal_gripper_action_controller_parameters.yaml:12]:
```suggestion
    read_only: true,
```

I would remove the default empty value for the `joint` parameter

**Review by saikishor — CHANGES_REQUESTED** [review]
I've left some comments from my end!
Thank you for the nice work!

**Review by bmagyar — CHANGES_REQUESTED** [review]
Reminder: We've agreed on renaming this to parallel_gripper_controller prior to merging

File: antipodal_gripper_controller/src/antipodal_gripper_action_controller_parameters.yaml
Comment (pac48) [reply antipodal_gripper_controller/src/antipodal_gripper_action_controller_parameters.yaml:29]: Yeah, basically the `max_velocity_interface` will be initialized to `max_velocity`. If the hardware does not support this, then it probably won't export the `max_velocity_interface` in the first place. And we do not limit this value, it is upp to the hardware driver to use this value.

Comment (saikishor) [reply antipodal_gripper_controller/src/antipodal_gripper_action_controller_parameters.yaml]: Got it. Make sense

File: ros2_controllers.rolling.repos
Comment (bmagyar) [inline ros2_controllers.rolling.repos:13]:
```suggestion
    url: https://github.com/ros-controls/control_msgs.git
    version: master
```

**Comment by nbbrooks** [comment]
@bmagyar It looks like the renaming to `parallel_gripper_controller` has been addressed - can you resolve your request for changes?

File: parallel_gripper_controller/include/parallel_gripper_controller/parallel_gripper_action_controller_impl.hpp
Comment (bmagyar) [inline parallel_gripper_controller/include/parallel_gripper_controller/parallel_gripper_action_controller_impl.hpp:327]: do we want to keep this?

**Review by bmagyar — APPROVED** [review]
Spotted one last piece of commented code, otherwise looks good!

File: parallel_gripper_controller/include/parallel_gripper_controller/parallel_gripper_action_controller_impl.hpp
Comment (bmagyar) [reply parallel_gripper_controller/include/parallel_gripper_controller/parallel_gripper_action_controller_impl.hpp:327]:
```suggestion
```

**Pull request merged**
